### DISCUSSION TITLE
Is Bun.serve faster than Bun node:http compatibility?

### DISCUSSION THREAD

**Discussion by whazor** [oven-sh/bun#9509] [General]
For a code base I am wondering it would be faster to use `Bun.serve`, and refactor away `node:http`. It would be a more significant undertaking as the code base also uses Connect middleware. Before doing so, I was wondering if anyone did some performance benchmarks between these two and whether there is a significant difference.

**Comment by Jarred-Sumner** [comment]
yes it is 2x - 3x faster to directly use Bun.serve(). In the future we plan to rewrite node:http to address this but we won't do that for awhile
